src/ray/core_worker/context.cc

```diff
@@ -0,0 +1,104 @@
+
+#include "context.h"
+
+namespace ray {
+
+/// per-thread context for core worker.
+struct WorkerThreadContext {
+  WorkerThreadContext(WorkerType worker_type, const DriverID &driver_id)
+    : task_index(0),
+      put_index(0) {
+
+    auto initial_driver_id = driver_id;
+    if (initial_driver_id.is_nil()) {
+      initial_driver_id = DriverID::from_random();
+    }
+  
+    if (worker_type == WorkerType::DRIVER) {
+      current_task_id = TaskID::from_random();
+    } else {
+      current_task_id = TaskID::nil();
+    }
+  }
+
+  int GetNextTaskIndex() {
+    return ++task_index;
+  }
+
+  int GetNextPutIndex() {
+    return ++put_index;
+  }
+
+  const DriverID &GetCurrentDriverID() {
+    return current_driver_id;
+  }
+
+  const TaskID &GetCurrentTaskID() {
+    return current_task_id;
+  }
+
+  void SetCurrentTask(const raylet::TaskSpecification &spec) {
+    current_driver_id = spec.DriverId();
+    current_task_id = spec.TaskId();
+    task_index = 0;
+    put_index = 0;
+  }
+
+  /// The driver ID for current task.
+  DriverID current_driver_id;
```

Comment:
`current_driver_id` shouldn't be part of `WorkerThreadContext`?

Reply:
right, removed.